sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -135,11 +160,12 @@
      * @throws NullPointerException if {@code keyCreateOptions} is {@code null}.
      * @throws ResourceModifiedException if {@code keyCreateOptions} is malformed.
      * @throws HttpRequestException if {@code name} is empty string.
-     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key created key}.
+     * @return A {@link Mono} containing the {@link Key created key}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Key>> createKey(KeyCreateOptions keyCreateOptions) {
-        return withContext(context -> createKey(keyCreateOptions, context));
+    public Mono<Key> createKey(KeyCreateOptions keyCreateOptions) {
+        return withContext(context -> createKey(keyCreateOptions, context))
+            .flatMap(keyResponse -> Mono.justOrEmpty(keyResponse.value()));
```

Comment:
Not a fan of all the same lines of code for flatMap. I would suggest moving this out to a utility method in azure-core, and then doing something like `.flatMap(FluxUtils::flattenResponse)`, where flattenResponse would be the code you have above.